### DISCUSSION TITLE
Custom selection for paths

### DISCUSSION THREAD

**Discussion by Ahmed092** [fabricjs/fabric.js#9864] [Q&A]
The default behavior:  when you select multiple free drawn paths with the selection box(canvas.selection=true), it selects paths based on  hitting anywhere within the bounding box.
The desired behavior: when you select multiple free drawn paths, it selects paths only when it hits the actual path not its bounding rectangle.

**Comment by ShaMan123** [comment]
perPixelTargetFind

**Reply by Ahmed092** [reply]
I have already tried perPixelTargetFind. Yes it enables pixel targeting but only when you just click on the path itself. What I want is selecting multiple paths with the cursor only when the selection box hits the pixels of the path. Is there still a way to use perPixelTargetFind to achieve this?

**Reply by asturur** [reply]
No, per pixel target find works only for point selections today.
Making it work for multiselection is currently not supported and not a quick fix

**Reply by Ahmed092** [reply]
Thanks! Any suggestions for workarounds or approximations?

**Reply by asturur** [reply]
I do not have any.
Let's change this in a change proposal where someone can slowly build an idea of how this can be implemented. may be slow to be be useful and for sure is not going to happen quickly unless somehow as a great plan in a pocket and wants to put it in writing

**Reply by asturur** [reply]
i'll edit the opening topic accordingly

**Comment by ShaMan123** [comment]
Ahh sorry I missed that you are looking for area selection.
I implemented it using bezier-js and #9371

**Reply by Ahmed092** [reply]
Thanks for replying! I am a litle bit familiar with bezier.js but I am new to fabricjs actually. Can you give me a more detailed insight? Would be much appreciated!

**Reply by ShaMan123** [reply]
bezier-js has https://pomax.github.io/bezierjs/#intersect-line
Then it is a matter of creating the Bezier instances and sending the correct line data depending on the plane of the Path.
To achieve that I suggest you send the 4 lines defining the area selection to the bezier plane.
That can be achieve as follows more or less:

```
let path = new fabric.Path(...)
const bezier = new Bezier(...)
const t = multiplyTransformMatrixArray([
      canvas.viewportTransform, // in case areaSelectionPoint is a viewport point
      path.calcTransformMatrix(),
      createTranslateMatrix(-path.pathOffset.x, -path.PathOffset.y)
   ])
const areaSelectionInBezierPlane = [tl, tr, br, bl].map(areaSelectionPoint => sendPointToPlane(
   areaSelectionPoint,
   undefined,
   t
))
// for each line do
[
   { p1: areaSelectionInBezierPlane.tl, p2: areaSelectionInBezierPlane.tr },
   { p1: areaSelectionInBezierPlane.tr, p2: areaSelectionInBezierPlane.br },
   { p1: areaSelectionInBezierPlane.br, p2: areaSelectionInBezierPlane.bl },
   { p1: areaSelectionInBezierPlane.bl, p2: areaSelectionInBezierPlane.tl },
].some(line => {
const intersectionTValues = bezier.intersects({
   p1: tl,
   p2: tr
})
const intersects = intersectionTValues.length > 0
return intersects
})
```
